internal/lib/stats/stats_server.go

```diff
@@ -228,3 +230,31 @@ func (ss *StatsServer) Shutdown() {
 	close(ss.shutdown)
 	ss.alreadyShutdown = true
 }
+
+// MetricsForPodSandbox returns the metrics for the given sandbox pod/container.
+func (ss *StatsServer) MetricsForPodSandbox(sb *sandbox.Sandbox) *SandboxMetrics {
+	ss.mutex.Lock()
+	defer ss.mutex.Unlock()
+	return ss.metricsForPodSandbox(sb)
+}
+
+// MetricsForPodSandboxList returns the metrics for the given list of sandboxes
+func (ss *StatsServer) MetricsForPodSandboxList(sboxes []*sandbox.Sandbox) []*SandboxMetrics {
+	ss.mutex.Lock()
+	defer ss.mutex.Unlock()
+	metricsList := make([]*SandboxMetrics, 0, len(sboxes))
+	for _, sb := range sboxes {
+		if metrics := ss.metricsForPodSandbox(sb); metrics != nil {
+			metricsList = append(metricsList, metrics)
+		}
+	}
+	return metricsList
+}
+
+// RemoveMetricsForPodSandbox removes the saved entry for the specified sandbox
+// to prevent the map from always growing.
```

Comment:
Mind you, given the internal implementation of Go's map, this does not stop it from growing. You don't have much control over the map's growth and when garbage will be collected unless you are very explicit about memory management.